.github/workflows/helm-rollback.yaml

```diff
@@ -0,0 +1,57 @@
+name: Helm Rollback
+on:
+  workflow_call:
+    inputs:
+      namespace:
+        description: "Kubernetes namespace to deploy resources in"
+        required: false
+        default: api
+        type: string
+      environment:
+        description: "Target environment for deployment (stg|prod)"
+        required: true
+        type: string
+      deployment:
+        description: "Name of the deployment to rollback. Defaults to repo name."
+        required: false
+        type: string
+    secrets:
+      k8s_server:
+        required: true
+      k8s_token:
+        required: true
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.image_tag }}
+      - name: Setup Helm
+        uses: azure/setup-helm@v3
+      - name: Setup Kubectl
+        uses: Azure/setup-kubectl@v3
+      - name: deploy
+        run: |
+          kubectl config set-cluster k8s --server="${{ secrets.k8s_server }}"
+          kubectl config set-credentials deploy --token="${{ secrets.k8s_token }}"
+          kubectl config set-context default --cluster=k8s --user=deploy
+          kubectl config use-context default
+          helm version
+          kubectl version
+          DEPLOYMENT="${{ github.event.repository.name }}"
+          if [ ! -z "${{ inputs.deployment }}" ]; then
+            DEPLOYMENT="${{ inputs.deployment }}"
+          fi
+          helm upgrade ${DEPLOYMENT} -n ${{ inputs.namespace }}
```

Comment:
is this supposed to be `helm rollback ...`

Reply:
Sigh.... my head is full of snot at the moment. Thanks.